debian/ceph-mgr-dashboard.requires

```diff
@@ -0,0 +1,12 @@
+bcrypt
+CherryPy
+more-itertools
+PyJWT
+pyopenssl
+requests
+Routes
+prettytable
+pytest
+pyyaml
+natsort
```

Comment:
> Actually `natsort` is used by the `orchestrator` module.

@rkachach i was referencing https://github.com/ceph/ceph/blob/4cb6436d6792cc8e82914ac141020746687bcd62/src/pybind/mgr/dashboard/requirements.txt#L12.  as i don't quite understand the rationale behind the change of 7e646d99589205633a21e19cbdc11b7999ae5da1 which introduced this dependency. i added it here as well.

if you believe that we don't need `natsort`, i will create a commit to drop it from `src/pybind/mgr/dashboard/requirements.txt`.

> I think it'd better to have it as a ceph-mgr-cephadm requirement rather than dashboard.

`orchestrator` module is included as part of `ceph-mgr-modules-core`. i am not sure why `natsort` should be a dependency of ` ceph-mgr-cephadm `. could you explain this in more details?

Reply:
> > Actually `natsort` is used by the `orchestrator` module.
> 
> @rkachach i was referencing
> 
> https://github.com/ceph/ceph/blob/4cb6436d6792cc8e82914ac141020746687bcd62/src/pybind/mgr/dashboard/requirements.txt#L12
> 
> . as i don't quite understand the rationale behind the change of [7e646d9](https://github.com/ceph/ceph/commit/7e646d99589205633a21e19cbdc11b7999ae5da1) which introduced this dependency. i added it here as well.
> if you believe that we don't need `natsort`, i will create a commit to drop it from `src/pybind/mgr/dashboard/requirements.txt`.
> 
> > I think it'd better to have it as a ceph-mgr-cephadm requirement rather than dashboard.
> 
> `orchestrator` module is included as part of `ceph-mgr-modules-core`. i am not sure why `natsort` should be a dependency of `ceph-mgr-cephadm`. could you explain this in more details?

Sure, `natsort` was originally introduced as an enhancement to address the tracker https://tracker.ceph.com/issues/54026. Right now, it's used by the orchestrator module:

`orchestrator/module.py:    from natsort import natsorted
`
The orchestrator is an `interface` that could be implemented by either `cephadm` or `rook` so both of them include this file to implement the interface. As far as I remember when I introduced the change the first time some tests on dashboard failed (because dashboard includes some file from orchestrator) that's why it was included the as dashboard dependency as well. 

I think with your changes probably it's  enough to have it only on `ceph-mgr-modules-core` (I'm assuming that dashboard will include this dependencies as well).

Reply:
yeah, `ceph-mgr-modules-core` is a hard dependency of `ceph-mgr`. but IMHO, it'd be nice to be explicit, say, put the direct dependencies as a packages' runtime dependencies. also, i wanted to mirror the dashboard's `requirements.txt` as i see this change as the first step to a unified solution to prepare the dependencies for both rpm and debian packaging. the discrepancies between 

- .spec
- `requirements.txt`
- `*.requires`

would be the recipe of headache of our posterities in long term. anyhow, i will drop it from `ceph-mgr-dashboard.requires` in my next revision. but the discrepancies still worry me.

Reply:
@rkachach please see https://github.com/ceph/ceph/blob/main/doc/dev/continuous-integration.rst#build-dependencies . probably you could expand that document based on our discussions and your findings.